Clover/app/src/main/java/org/floens/chan/ui/captcha/v2/CaptchaNoJsLayoutV2.java

```diff
@@ -0,0 +1,274 @@
+        captchaImagesGrid = view.findViewById(R.id.captcha_layout_v2_images_grid);
+        captchaVerifyButton = view.findViewById(R.id.captcha_layout_v2_verify_button);
+        useOldCaptchaButton = view.findViewById(R.id.captcha_layout_v2_use_old_captcha_button);
+        reloadCaptchaButton = view.findViewById(R.id.captcha_layout_v2_reload_button);
+        buttonsHolder = view.findViewById(R.id.captcha_layout_v2_buttons_holder);
+        background = view.findViewById(R.id.captcha_layout_v2_background);
+
+        background.setBackgroundColor(AndroidUtils.getAttrColor(getContext(), R.attr.backcolor));
+        buttonsHolder.setBackgroundColor(AndroidUtils.getAttrColor(getContext(), R.attr.backcolor_secondary));
+        captchaChallengeTitle.setBackgroundColor(AndroidUtils.getAttrColor(getContext(), R.attr.backcolor_secondary));
+        captchaChallengeTitle.setTextColor(AndroidUtils.getAttrColor(getContext(), R.attr.text_color_primary));
+        captchaVerifyButton.setTextColor(AndroidUtils.getAttrColor(getContext(), R.attr.text_color_primary));
+        useOldCaptchaButton.setTextColor(AndroidUtils.getAttrColor(getContext(), R.attr.text_color_primary));
+        reloadCaptchaButton.setTextColor(AndroidUtils.getAttrColor(getContext(), R.attr.text_color_primary));
+
+        captchaVerifyButton.setOnClickListener(this);
+        useOldCaptchaButton.setOnClickListener(this);
+        reloadCaptchaButton.setOnClickListener(this);
+    }
+
+    @Override
+    protected void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+
+        reset();
+    }
+
+    @Override
+    public void initialize(Site site, AuthenticationLayoutCallback callback) {
+        this.callback = callback;
+
+        SiteAuthentication authentication = site.actions().postAuthenticate();
+        if (authentication.type != SiteAuthentication.Type.CAPTCHA2_NOJS) {
+            callback.onFallbackToV1CaptchaView();
+            return;
+        }
+
+        presenter.init(authentication.siteKey, authentication.baseUrl);
+    }
+
+    @Override
+    public void reset() {
+        hardReset();
+    }
+
+    @Override
+    public void hardReset() {
+        CaptchaNoJsPresenterV2.RequestCaptchaInfoError captchaInfoError = presenter.requestCaptchaInfo();
+
+        switch (captchaInfoError) {
+            case Ok:
+                break;
+            case HoldYourHorses:
+                showToast(getContext().getString(
+                        R.string.captcha_layout_v2_you_are_requesting_captcha_too_fast));
+                break;
+            case AlreadyInProgress:
+                showToast(getContext().getString(
+                        R.string.captcha_layout_v2_captcha_request_is_already_in_progress));
+                break;
+        }
+    }
+
+    @Override
+    public void onCaptchaInfoParsed(CaptchaInfo captchaInfo) {
+        // called on a background thread
+
+        AndroidUtils.runOnUiThread(() -> {
```

Comment:
Should probably disable the Verify button until we have the CaptchaInfo. Do not forget to reset the button's disabled state in initialize/onError/onVerificationDone/onDestroy.